### PR TITLE
BACK-1075: fix pancakeswapv3 pricing

### PR DESCRIPTION
**Pull request by paraschiv-razvan-toptal**

### CODE REVIEW

File: src/dex/pancakeswap-v3/config.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    [SwapSide.BUY]: [{ name: 'ArbitrumBuyAdapter', index: 2 }],
+  },
+  [Network.OPTIMISM]: {
+    [SwapSide.SELL]: [{ name: 'OptimismAdapter01', index: 3 }],
+    [SwapSide.BUY]: [{ name: 'OptimismBuyAdapter', index: 2 }],
+  },
+  [Network.FANTOM]: {
+    [SwapSide.SELL]: [{ name: 'FantomAdapter01', index: 11 }],
+    [SwapSide.BUY]: [{ name: 'FantomBuyAdapter', index: 3 }],
+  },
```
Comment (Verisana) [inline src/dex/pancakeswap-v3/config.ts:66]: If protocol is not deployed on these chains, better to remove this

File: src/dex/pancakeswap-v3/constants.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+export const PANCAKESWAP3_FUNCTION_CALL_GAS_COST = 21_000; // Ceiled
+export const PANCAKESWAPV3_TICK_GAS_COST = 24_000; // Ceiled
```
Comment (Verisana) [inline src/dex/pancakeswap-v3/constants.ts:2]: I think it should be adapted to recent UniV3 changes

File: src/dex/pancakeswap-v3/config.ts (hunk 1 above)
Comment (aburkut) [reply src/dex/pancakeswap-v3/config.ts]: @Verisana updated

File: src/dex/pancakeswap-v3/constants.ts (hunk 2 above)
Comment (aburkut) [reply src/dex/pancakeswap-v3/constants.ts]: @Verisana updated

**Review by aburkut — APPROVED** [review]

**Review by aburkut — APPROVED** [review]

**Review by Verisana — APPROVED** [review]

**Pull request merged**